buidl/script.py

```diff
@@ -224,6 +230,39 @@ def evaluate(self, z, witness):
                     stream = BytesIO(encode_varstr(witness_script))
                     witness_script_commands = Script.parse(stream).commands
                     commands.extend(witness_script_commands)
+                # witness program version 1 rule. if stack commands are:
+                # 0 <32 byte hash> this is p2tr
+                elif len(stack) == 2 and stack[0] == b"\x01" and len(stack[1]) == 32:
+                    if len(witness) == 0:
+                        print("stack in witness v1 empty")
+                        return False
+                    if len(witness) > 1 and witness[-1][0] == 0x50:
+                        # this is the annex, which is unused
+                        witness.items.pop()
+                    if len(witness) == 1:
+                        # this is a key path spend
+                        sig = witness[0]
+                        # pubkey is already at stack[1], so assign sig
+                        stack[0] = sig
+                        op_checksig_schnorr(stack, sig_hash_function)
+                    elif len(witness) > 1:
+                        from buidl.taproot import ControlBlock, TapLeaf
```

Comment:
No way to avoid this?

Reply:
It's a circular import. buidl.taproot imports P2TRScriptPubKey and script needs this stuff from taproot.